Adapter/Configuration.php

```diff
@@ -25,8 +25,22 @@
  */
 namespace PrestaShop\PrestaShop\Adapter;
 
-class Configuration implements \PrestaShop\PrestaShop\Core\ConfigurationInterface
+use PrestaShop\PrestaShop\Core\Foundation\Exception;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
 {
+    private $shop;
+
+    public function __construct(\Shop $shop = null)
+    {
+        if (is_null($shop)) {
```

Comment:
Because of the type hinting, `Shop` can be either null or `instanceof Shop`, so no, checking whether it is an instance of `Shop` is useless.